File: web-service/src/main/java/enrole/UserRole.java
package enrole;

/**
 *
 * @author marcw
 */
public class UserRole {

	String name, description;

	public static enum PermissionItme {
		USER, ASSIGNMENT, ASSIGNMENT_SUBMIT, COURSE, COURSE_OFFERING, DEPARTMENT, GROUP
	}

	public static class Permission {

		PermissionItme Itme;
		boolean create, read, update, delete;
	}
}

File: front-end/src/app/models/user/user-role.ts
import { Permission } from './permission';
import { SEntity } from '../core/s-entity';

/**
 * Holds the user permissions and access
 * @author Marc Wafik
 */
export class UserRole extends SEntity {

	name: string;
	description: string;
	isStaff: boolean;
	canEditOptions: boolean;

	permissions: {
		UserRole: Permission,
		User: Permission,
		Department: Permission,
		Course: Permission,
		Semester: Permission,
		CourseOffering: Permission,
		Group: Permission,
		SemesterEnrollment: Permission,
		CourceOfferingEnrollment: Permission
	};

	constructor() {
		super();
		this.name = "";
		this.description = "";
		this.isStaff = false;
		this.canEditOptions = false;

		this.permissions = {
			UserRole: new Permission(),
			User: new Permission(),
			Department: new Permission(),
			Course: new Permission(),
			Semester: new Permission(),
			CourseOffering: new Permission(),
			Group: new Permission(),
			SemesterEnrollment: new Permission(),
			CourceOfferingEnrollment: new Permission()
		};
	}

}